themes/src/main/resources/theme/base/admin/resources/js/app.js

```diff
@@ -3270,6 +3270,21 @@ module.directive('kcValidPage', function() {
    }
 });
 
+// Directive to parse/format strings into numbers
+module.directive('stringToNumber', function() {
```

Comment:
@mhajas If I remove the `string-to-number` from the input and set any valid number (30 for example). After saving, the field is empty and the JS console (in chrome) says:

```
angular.js:14961 Error: [ngModel:numfmt] http://errors.angularjs.org/1.6.10/ngModel/numfmt?p0=13
    at angular.js:88
    at Array.<anonymous> (angular.js:25862)
    at Object.$$format (angular.js:30061)
    at Object.$processModelValue (angular.js:30041)
    at Object.$$setModelValue (angular.js:30073)
    at angular.js:30112
    at m.$digest (angular.js:18535)
    at m.$apply (angular.js:18903)
    at k (angular.js:12775)
    at s (angular.js:13032)
```
Check [the issue I link before](https://docs.angularjs.org/error/ngModel/numfmt).

Reply:
Ah I see it now. Sorry. So to fully understand it, when you not include the directive, it fails because angular is trying to store an integer into identityProvider.config.allowedClockSkew which is supposed to be a string?

Reply:
The problem is reading the value from the json. It fails because it expects an integer and it's a string (the value that comes from the endpoint is a string, "60" for example). So it complains with that error and sets nothing (the input is empty). The saving is OK, but the read fails.

I think that the OIDC used a text input because of this.

Yes, I'm right. I have just tested with the `$formaters` function and it works:

```javascript
module.directive('stringToNumber', function() {
    return {
        require: 'ngModel',
        link: function(scope, element, attrs, ngModel) {
            //ngModel.$parsers.push(function(value) {
            //    return (typeof value === 'undefined' || value === null)? '' : '' + value;
            //});
            ngModel.$formatters.push(function(value) {
                return parseFloat(value);
            });
        }
    };
});
```

So, if you want I can remove the commented part.

Reply:
I like the approach via the directive due to min / max limits. Since this affects app.js, I'd like to ask @vmuzikar to review this part of the code.

Reply:
Only this one remains. I like using an input because the javascript controls any invalid character in the number (or decimals, or whatever). Okis, I have submitted the rest of the changes and this point is the same for the moment.

Reply:
@rmartinc Thank you for your contribution!

@hmlnarik Even though this seems to me like a significant overkill (using a plain input field would do just fine – we're going to rewrite the whole admin console soon anyway), I see nothing wrong with this approach – seems to be the usual solution for problems like this. And it even works with IE. :D

LGTM.

Reply:
Thanks @vmuzikar!